submissions/VikTorNik/froggen-oojs/app.js

```diff
@@ -0,0 +1,122 @@
+const GAME_CONF = {
+    OffsetX: 101,
+    OffsetY: 83,
+    overlap: 0.8,
+    collumn: 5,
+    countWinner: 0,
+    countСheckmate: 0,
+};
+
+const PLAYER_CONF = {
+    startX: 202,
+    startY: 405,
+    offsetMinX: 0,
+    offsetMaxX: 400,
+    offsetMinY: 0,
+    offsetMaxY: 400,
+    sprite: "images/char-boy.png",
+};
+
+const ENEMY_CONF = {
+    startY: GAME_CONF.OffsetY / 4,
+    speed: 50,
+    maximumCount: 5,
+    sprite: 'images/enemy-bug.png',
+};
+
+const Elementary = function (x, y, sprite) {
+    this.x = x;
+    this.y = y;
+    this.sprite = sprite;
+}
+
+const Enemy = function (x, y, sprite, speed, player) {
+    Elementary.call(this, x, y, sprite);
+    this.speed = speed;
+    this.player = player;
+}
+// Enemy.prototype = Object.create(Elementary.prototype);
+// Enemy.prototype.constructor = Enemy;
+
+Enemy.prototype.update = function (dt) {
+    this.x += this.speed * dt;
+    this.x > GAME_CONF.OffsetX * GAME_CONF.collumn ? this.x = -GAME_CONF.OffsetX : false;
+    this.collision();
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.collision = function () {
+    ((this.player.x < this.x + GAME_CONF.OffsetX * GAME_CONF.overlap) &&
+        (this.player.x > this.x - GAME_CONF.OffsetX * GAME_CONF.overlap) &&
+        (this.player.y < this.y + GAME_CONF.OffsetY * GAME_CONF.overlap) &&
+        (this.player.y > this.y - GAME_CONF.OffsetY * GAME_CONF.overlap)) ?
+        (this.player.x = PLAYER_CONF.startX,
+            this.player.y = PLAYER_CONF.startY,
+            GAME_CONF.countСheckmate += 1)
+        : false;
+};
+
+const Player = function (x, y, sprite) {
+    Elementary.call(this, x, y, sprite);
+}
+
+Player.prototype.update = function () {
+    this.offsetMinX = PLAYER_CONF.offsetMinX;
+    this.offsetMaxX = PLAYER_CONF.offsetMaxX;
+    this.offsetMinY = PLAYER_CONF.offsetMinY;
+    this.offsetMaxY = PLAYER_CONF.offsetMaxY;
+    this.winner();
+};
+
+Player.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    this.countWinnerСheckmate();
+};
+
+Player.prototype.winner = function () {
+    this.y < 0
+        ? (this.x = PLAYER_CONF.startX, this.y = PLAYER_CONF.startY, GAME_CONF.countWinner += 1)
+        : false;
```

Comment:
Please explain this. The result of ternary is never used. What is the purpose?

Reply:
Using the ternary operator is more concise and more concise. I like this option better than using "if".

```
    if (this.y < 0) {
        this.x = PLAYER_CONF.startX;
        this.y = PLAYER_CONF.startY;
        GAME_CONF.countWinner += 1;
    }
```

Reply:
We write code for other people. Computers are ok with byte-codes.
The purpose of a ternary operator is to provide a value. `false` that is never used creates feeling of incompleteness.